ui/src/plugins/dev.perfetto.LargeScreensPerf/index.ts

```diff
@@ -13,32 +13,101 @@
 // limitations under the License.
 
 import {Trace} from '../../public/trace';
+import {TrackNode} from '../../public/workspace';
 import {PerfettoPlugin} from '../../public/plugin';
+import {NUM} from '../../trace_processor/query_result';
 
 export default class implements PerfettoPlugin {
   static readonly id = 'dev.perfetto.LargeScreensPerf';
   async onTraceLoad(ctx: Trace): Promise<void> {
     ctx.commands.registerCommand({
-      id: 'dev.perfetto.LargeScreensPerf#PinUnfoldLatencyTracks',
-      name: 'Pin: Unfold latency tracks',
-      callback: () => {
-        ctx.workspace.flatTracks.forEach((track) => {
-          if (
-            !!track.name.includes('UnfoldTransition') ||
-            track.name.includes('Screen on blocked') ||
-            track.name.includes('hingeAngle') ||
-            track.name.includes('UnfoldLightRevealOverlayAnimation') ||
-            track.name.startsWith('waitForAllWindowsDrawn') ||
-            track.name.endsWith('UNFOLD_ANIM>') ||
-            track.name.endsWith('UNFOLD>') ||
-            track.name == 'Waiting for KeyguardDrawnCallback#onDrawn' ||
-            track.name == 'FoldedState' ||
-            track.name == 'FoldUpdate'
-          ) {
-            track.pin();
-          }
-        });
+      id: 'dev.perfetto.LargeScreensPerf#UnfoldLatencyTracks',
+      name: 'Organize unfold latency tracks',
+      callback: async () => {
+        this.pinCoreTracks(ctx);
+        this.addUnfoldMiscSection(ctx);
+        await this.addUnfoldDisplaySwitchingSection(ctx);
+        this.addUnfoldAnimationSection(ctx);
       },
     });
   }
+
+  private pinCoreTracks(ctx: Trace) {
+    const titlesToPin = new Set(['FoldedState', 'FoldUpdate']);
+    ctx.workspace.flatTracks
+      .filter(
+        (track) =>
+          titlesToPin.has(track.name) ||
+          track.name.includes('hingeAngle') ||
+          track.name.endsWith('UNFOLD>'),
+      )
+      .forEach((track) => track.pin());
+  }
+
+  private addUnfoldMiscSection(ctx: Trace) {
+    // section for tracks that don't fit neatly in other sections and are not so important to be pinned
+    const group = new TrackNode({name: 'Unfold misc'});
```

Comment:
can we have all of these as children of a top level unfold group?

Actually.. I wonder if we should make this always enabled: if this only adds an "Folded state" top level track group (that by default in its track shows the folded state e.g. folded/unfolded), and then as it's expanded it shows the other groups, it would be quite useful and probably not an issue to have it always enabled